libvast/include/vast/system/spawn_arguments.hpp

```diff
@@ -47,10 +46,15 @@ struct spawn_arguments {
   }
 
   template <class Inspector>
-  friend auto inspect(Inspector& f, spawn_arguments& x) ->
-    typename Inspector::result_type {
-    return f(caf::meta::type_name("vast.system.spawn_arguments"), x.inv, x.dir,
-             x.label, x.expr);
+    requires(Inspector::is_loading == false)
+  friend auto inspect(Inspector& f, spawn_arguments& x) {
+    auto inv = x.inv;
+    auto dir = x.dir.string();
+    auto label = x.label;
```

Comment:
All of these are const references. The inspector api requires them to be non const so the easy way is just to create copies locally. Seems to be only used for serialization

Reply:
This is a good approach I think. This is definitely not in the hot path and even then this would just be copying a few strings.